google/cloud/storage/internal/rest/stub.cc

```diff
@@ -165,8 +179,7 @@ StatusOr<ListBucketsResponse> RestStub::ListBuckets(
     ListBucketsRequest const& request) {
   RestRequestBuilder builder(
       absl::StrCat("storage/", options.get<TargetApiVersionOption>(), "/b"));
-  auto auth = AddAuthorizationHeader(options, builder);
-  if (!auth.ok()) return auth;
+  AddHeaders(options, builder);
```

Comment:
You probably should be checking the status returned from AddHeaders just like it was doing for AddAuthorizationHeader.